### PR TITLE
TangoShutter (dual command - read-write attribute)

### PR DESCRIPTION
**Pull request by vrey01**
The main goal of this commit is to change TangoShutter to either use commands configured in xml (as before) or if those are not configure, use the State channel as a read/write attribute.

### CODE REVIEW

File: HardwareObjects/TangoShutter.py [hunk 1]
```diff
@@ -1,90 +1,215 @@
...
-        #
-        self.shutterStateValue = value
-        # print "emit : ",TangoShutter.shutterState[self.shutterStateValue]
-        self.emit('shutterStateChanged', (self.getShutterState(),))
+        self.state_value_str = self._convert_state_to_str(value)
+        self.emit('shutterStateChanged', (self.state_value_str,))
 
+    def _convert_state_to_str(self, value):
+        state = str(value).upper()
+        if state in self.shutterState:
```
Comment (mguijarr) [inline HardwareObjects/TangoShutter.py]: Could be `state = self.shutterState.get(state, "unknown")`

File: HardwareRepository.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def require(self, mnemonicsList):
             mnemonics = ",".join([repr(mne) for mne in mnemonicsList])
             if len(mnemonics) > 0:
                 self.requiredHardwareObjects = SpecWaitObject.waitReply(self.server, 'send_msg_cmd_with_return' , ('xml_getall(%s)' % mnemonics, ), timeout = 3)
-                logging.getLogger("HWR").debug("Getting all the hardware objects took %s ms." % ((time.time()-t0)*1000))
+                logging.getLogger("HWR").debug("Getting all the hardware objects (%s) took %s ms." % (len(self.requireHardwareObjects), (time.time()-t0)*1000))
```
Comment (mguijarr) [inline HardwareRepository.py]: Would "Getting %s hardware objects took %s ms." be better?

**Comment by mguijarr** [comment]
Thanks for this :)

I made 2 little comments, a bit pedantic I have to say ! So if you have time to change according to the comments it's perfect otherwise I will merge the code tomorrow at least in anyways.

**Comment by vrey01** [comment]
Both comments adopted. :-)

**Pull request merged**